src/decision_model.jl

```diff
@@ -348,3 +365,286 @@ function DecisionStrategy(diagram::InfluenceDiagram, z::OrderedDict{Name, Decisi
+    for index in CartesianIndices(μ_bar_statevars)
+        for s_j in 1:length(diagram.States[name])
+            if isa(diagram.Nodes[name], ChanceNode)
+                # μ_{C_v} = μ_{\bar{C}_v}*p
+                @constraint(model, μ_statevars[Tuple(index)...,s_j] == diagram.X[name][Tuple(index)[I_j_mapping_in_cluster]...,s_j]*μ_bar_statevars[index])
+            elseif isa(diagram.Nodes[name], DecisionNode)
+                # μ_{C_v} ≤ z
+                @constraint(model, μ_statevars[Tuple(index)...,s_j] <= z[name].z[Tuple(index)[I_j_mapping_in_cluster]...,s_j])
+            end
+        end
+    end
+end
+
+
+struct RJTVariables
+    data::Dict{Name, μVariable}
+end
+
+"""
+    RJTVariables(model, diagram, z)
+
+Using the influence diagram and decision variables z from DecisionProgramming.jl, adds the 
+variables and constraints of the corresponding RJT model.
+
+# Arguments
+- `model::Model`: JuMP model into which variables are added.
+- `diagram::InfluenceDiagram`: Influence diagram structure.
+- `z::OrderedDict{Name, DecisionVariable}`: Decision variables from `DecisionVariables` function.
+
+# Examples
+```julia
+μ_s = RJTVariables(model, diagram, z)
+```
+"""
+function RJTVariables(model::Model, diagram::InfluenceDiagram, z::OrderedDict{Name, DecisionVariable})
+    # Get the RJT structure
+    # SHOULD WE DEFINE C_rjt and A_rjt AS PART OF THE INFLUENCE DIAGRAM STRUCT OR AS THEIR OWN, MAYBE BETTER TO DEFINE AS PART OF INFLUENCE DIAGRAM IF THERE IS NEVER A NEED
+    # TO HAVE AN INDEPENDENT RJT NOT LINKED TO AN INFLUENCE DIAGRAM
+    diagram.C_rjt, diagram.A_rjt = ID_to_RJT(diagram)
+
+    #μ- AND μ_bar-VARIABLES IN THE SAME LOOP, OTHERWISE NOT A PROBLEM, BUT DIFFERENT ORDER COMPARED TO PAPER AND LIKEWISE SLIGHTLY WEIRD ORDER OF EQUATIONS IN MODEL PRINTOUT
+    # Variables corresponding to the nodes in the RJT
+    μVars = Dict{Name, μVariable}()
+    # Variables μ_{\bar{C}_v} = ∑_{x_v} μ_{C_v}
+    μBarVars = Dict{Name, μVariable}()
+    for name in union(keys(diagram.C), keys(diagram.D))
+    #for name in union(keys(diagram.C), keys(diagram.D), keys(diagram.V))
+        μVars[name] = μVariable(name, μ_variable(model, name, diagram.States, diagram.C_rjt))
+        μBarVars[name] = μVariable(name, μ_bar_variable(model, name, diagram.States, diagram.C_rjt, μVars[name].statevars))
+    end
+
+    # Enforcing local consistency between clusters, meaning that for a pair of adjacent clusters, 
+    # the marginal distribution for nodes that are included in both, must be the same.
+    for arc in diagram.A_rjt
+        if !isa(diagram.Nodes[arc[2]], ValueNode)
+            local_consistency_constraint(model, arc, diagram.C_rjt, μVars)
+        end
+    end
+
+    # Add in the conditional probabilities and decision strategies
+    # In our structure, value nodes are not stochastic and the objective function doesn't really work in this context. 
```

Comment:
"objective function doesn't really work" is a bit weird and vague (might be written by me). I think we could just remove "and the objective function doesn't really work in this context."

Reply:
Fixed comment